src/main/java/kr/co/vcnc/haeinsa/HaeinsaTable.java

```diff
@@ -233,21 +234,21 @@ public HaeinsaResultScanner getScanner(@Nullable HaeinsaTransaction tx, HaeinsaS
         HaeinsaTableTransaction tableState = tx.createOrGetTableState(getTableName());
         NavigableMap<byte[], HaeinsaRowTransaction> rows;
 
-        if (Bytes.equals(scan.getStartRow(), HConstants.EMPTY_START_ROW)) {
-            if (Bytes.equals(scan.getStopRow(), HConstants.EMPTY_END_ROW)) {
+        if (Bytes.equals(!scan.isReversed() ?  scan.getStartRow() : scan.getStopRow(), HConstants.EMPTY_START_ROW)) {
+            if (Bytes.equals(!scan.isReversed() ? scan.getStopRow() : scan.getStartRow(), HConstants.EMPTY_END_ROW)) {
                 // null, null
                 rows = tableState.getRowStates();
             } else {
                 // null, StopRow
-                rows = tableState.getRowStates().headMap(scan.getStopRow(), false);
+                rows = tableState.getRowStates().headMap(!scan.isReversed() ? scan.getStopRow() : scan.getStartRow(), false);
             }
         } else {
-            if (Bytes.equals(scan.getStopRow(), HConstants.EMPTY_END_ROW)) {
+            if (Bytes.equals(!scan.isReversed() ? scan.getStopRow() : scan.getStartRow(), HConstants.EMPTY_END_ROW)) {
                 // StartRow, null
-                rows = tableState.getRowStates().tailMap(scan.getStartRow(), true);
+                rows = tableState.getRowStates().tailMap(!scan.isReversed() ? scan.getStartRow() : scan.getStopRow(), true);
             } else {
                 // StartRow, StopRow
-                rows = tableState.getRowStates().subMap(scan.getStartRow(), true, scan.getStopRow(), false);
+                rows = tableState.getRowStates().subMap(!scan.isReversed() ? scan.getStartRow() : scan.getStopRow(), true, !scan.isReversed() ? scan.getStopRow() : scan.getStartRow(), false);
```

Comment:
I wonder if it would be useful and less error-prone if we factored these out to like:

``` java
byte[] begin = !scan.isReversed() ?  scan.getStartRow() : scan.getStopRow();
byte[] end = !scan.isReversed() ? scan.getStopRow() : scan.getStartRow();
```

Or maybe better even to have `HaeinsaScan` itself return the correct one from its `scan.getStartRow()`/`scan.getStopRow()` methods since it knows whether it's reversed or not, so something like `HaeinsaTable` doesn't need to be aware of its implementation details.

Reply:
I will do 
```
byte[] begin = !scan.isReversed() ?  scan.getStartRow() : scan.getStopRow();
byte[] end = !scan.isReversed() ? scan.getStopRow() : scan.getStartRow();
```. 
I want to keep the convention like a native hbase scan.